[Bugfix] Gracefully error wolfram when no pods return

onerandomusername opened the pull request:
# Bugfix

As of now, location based queries to wolfram alpha error because the ip address provided is empty.
![image](https://user-images.githubusercontent.com/71233171/115802192-6f2f3800-a3ac-11eb-8df4-b062bed2d7fd.png)

This solves that issue by providing a mock ip address of cloudfare's dns servers, 1.1.1.1, and by giving a latitude and longitude of 0 degrees.
![image](https://user-images.githubusercontent.com/71233171/115802183-676f9380-a3ac-11eb-9aa0-a8ec81df7ca1.png)

onerandomusername committed:
add mock location and ip to wolfram

SimplyMika commented:
I don't think this fix is a good idea. You can already specify a location (`.w next full moon in New York`) and specifying no location is pretty meaningless anyway. Defaulting to some random location is unintuitive and misleading, imo.
I think it should get an improved error message. Catch the exception, see if the error was no IP Address / location specified, then show a reasonable error message.

onerandomusername commented:
> I don't think this fix is a good idea. You can already specify a location (`.w next full moon in New York`) and specifying no location is pretty meaningless anyway. Defaulting to some random location is unintuitive and misleading, imo.
I think it should get an improved error message. Catch the exception, see if the error was no IP Address / location specified, then show a reasonable error message. 

The reason to provide the mock parameters is to let every search work, even if the results aren't fully accurate. Upon further thought, what about a combination of both? Providing 0.0 for lat and long is taken as a default on most map apps when they don't have access to your location.

I do agree that its possible for the api to have no pods in the result, which does mean that error should be gracefully handled.

onerandomusername committed:
revert location fix, use graceful error for no pods

SimplyMika commented:
I don't think each search should work, that's the thing. Asking “when is the next full moon?” is like asking “how far is it to the next airport?”, there is no useful answer here if WolframAlpha doesn't know your location. Yes, you can estimate a location, and in the case of moon phases that may be accurate enough, but for quite a lot of things, this is pretty meaningless.

The better error message, however, is definitely a good idea probably for even more cases than just this.

onerandomusername commented:
I've added a check so in the event that there aren't any pods, the current wolfram error message is shown.
I've also reverted the location override.
![image](https://user-images.githubusercontent.com/71233171/116702660-5a980480-a997-11eb-9e08-586ab78a5847.png)

onerandomusername committed:
make error message more specific

onerandomusername committed:
fix location reason always being shown on error

Run1e commented on cogs/fun.py:
```diff
@@ -202,25 +203,34 @@ async def wolfram(self, ctx, *, query):
 			res = j['queryresult']
 
-			success = res['success']
+			success = bool(res['success'] and res['numpods'])
```
Wrapping in bool(...) is not necessary since `x and y` returns a boolean value already

Run1e commented on cogs/fun.py:
```diff
@@ -202,25 +203,34 @@ async def wolfram(self, ctx, *, query):
-					e.add_field(name='Tips from Wolfram Alpha:', value=res['tips']['text'])
+					e.add_field(name='Tips from Wolfram Alpha:', value=res['tips']['text'], inline=False)
+
+				if len(e.fields) == 0 and res['numpods'] == 0:
```
`len(e.fields) == 0` can be replaced with `e.fields`

Run1e commented on cogs/fun.py:
```diff
@@ -202,25 +203,34 @@ async def wolfram(self, ctx, *, query):
+
+				if len(e.fields) == 0 and res['numpods'] == 0:
+					e.add_field(name='Possible Reason:',
+									value=(
```
Funky indents

onerandomusername committed:
unwrap bool, simplify bool, fix formatting
